testenv/kubernetes/kind/dockerimageloader.go

```diff
@@ -52,7 +52,27 @@ func (l *DockerImageLoader) imageIDsFromRepoTags(ctx context.Context, repoTags [
 		}
 
 		if len(result) <= 0 {
-			return nil, fmt.Errorf("failed to find image: %s", repoTag)
+			// Pull image if not found locally
+			resp, err := l.docker.ImagePull(ctx, repoTag, imagetypes.PullOptions{})
+			if err != nil {
+				return nil, fmt.Errorf("failed to pull image %s: %w", repoTag, err)
+			}
+
+			// Drain response to avoid blocking
+			_, _ = io.Copy(io.Discard, resp)
+			_ = resp.Close()
+
+			result, err = l.docker.ImageList(ctx, imagetypes.ListOptions{
+				Filters: filters.NewArgs(
+					filters.Arg("reference", repoTag)),
+			})
+			if err != nil {
+				return nil, fmt.Errorf("failed to list image %s: %w", repoTag, err)
+			}
+
+			if len(result) <= 0 {
+				return nil, fmt.Errorf("failed to find image: %s", repoTag)
+			}
```

Comment:
we can continue here meaning that `imageIDsFromRepoTags` can return empty slice. we can move the responsibility to pull image to `Load` method from line 100

```go
		ids, err := l.imageIDsFromRepoTags(ctx, []string{image})
		if err != nil {
			return fmt.Errorf("failed to get ImageID for RepoTag %s: %w", image, err)
		}
		if len(ids) == 0 {
			// pull logic
			// ids = from pull info
		}
```

Reply:
lets also log when we pull the image